external-crates/move/crates/move-cli/src/base/new.rs

```diff
@@ -47,6 +49,14 @@ impl New {
     ) -> anyhow::Result<()> {
         // TODO warn on build config flags
         let Self { name } = self;
+
+        if !Identifier::is_valid(&name) {
+            return Err(anyhow!(
+                "invalid package name, please use lowercase letters, numbers, and underscores, \
+should start with a lowercase letter."
```

Comment:
Some formatting nits

```suggestion
                "Invalid package name, please use lowercase letters, numbers, and underscores, \
                 should start with a lowercase letter"
```